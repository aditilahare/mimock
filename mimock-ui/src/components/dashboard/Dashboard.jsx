import React from 'react';
import List from './List';
import Sidebar from './Sidebar';
import Titlebar from './Titlebar';
import Toolbar from './Toolbar';

export default function Dashboard() {
	return (
		<>
			<Titlebar />
			<Toolbar />
			<Sidebar />
			<List />
		</>
	);
}
