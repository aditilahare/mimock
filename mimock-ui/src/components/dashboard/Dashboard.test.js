import React from 'react';
import { render } from '@testing-library/react';
import Dashboard from './Dashboard';

describe('Dashboard', () => {
	it('should render all dashboard components', async () => {
		const tree = await render(<Dashboard />);

		const { container, getByTestId } = tree;

		expect(getByTestId('titlebar-section')).toBeInTheDocument();
		expect(getByTestId('toolbar-section')).toBeInTheDocument();
		expect(getByTestId('sidebar-section')).toBeInTheDocument();
		expect(getByTestId('list-section')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
