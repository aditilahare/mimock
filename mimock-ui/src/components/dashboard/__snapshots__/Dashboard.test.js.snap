// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render all dashboard components 1`] = `
<div>
  <div
    class="w-11/12 flex flex-row mx-auto mt-4 mb-2 justify-center align-middle"
    data-testid="titlebar-section"
  >
    <div
      class="font-sans text-2xl font-bold text-gray-700 mx-auto my-auto select-none"
    >
      <span
        class="inline-flex items-center justify-center"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
          />
        </svg>
         
        Mocks
      </span>
    </div>
    <input
      class="w-5/6 my-2 mx-4 p-2 outline-none rounded-md border-2 border-gray-300 bg-white placeholder:italic placeholder:text-slate-400 sm:text-sm"
      placeholder="Search"
    />
  </div>
  <hr />
  <div
    class="w-11/12 flex flex-row mx-auto mt-4 mb-2 justify-center align-middle"
    data-testid="toolbar-section"
  >
    <button
      class="flex items-center justify-center gap-3 px-9 py-3 m-auto shadow-md rounded-md text-center text-lg font-semibold cursor-pointer hover:brightness-105 text-gray-600 w-fit bg-gray-300 dark:bg-slate-300"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
        />
      </svg>
      Filter Mock
    </button>
    <button
      class="flex items-center justify-center gap-3 px-9 py-3 m-auto shadow-md rounded-md text-center text-lg font-semibold cursor-pointer hover:brightness-105 text-lime-700 dark:text-white w-fit bg-green-300 dark:bg-green-400"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 2.013H9V9H2v6h7v6.987h6V15h7V9h-7z"
        />
      </svg>
      Add Mock
    </button>
  </div>
  <hr />
  <div
    data-testid="sidebar-section"
  >
    Sidebar
  </div>
  <div
    data-testid="list-section"
  >
    List
  </div>
</div>
`;
